### ISSUE TITLE
Adding "count_include_pad" options to AvgPool layer

### ISSUE THREAD

**Issue by dslisleedh** [google/flax#2448]
as metioned in https://github.com/google/flax/issues/2401
Flax's pool layer divide window sum result including padded token but does not provide Pytorch options such as "count include pad".

However, since I needed this function to implement Poolformer(https://arxiv.org/abs/2111.11418) with flax, I made it myself. Hope this helps.

    class AvgPool2D(nn.Module):
        pool_window: int = 3
        strides: int = 3
        padding: str = 'SAME'
        count_include_pad: bool = True
    
        @nn.compact
        def __call__(self, x):
            _, h, w, _ = x.shape
            x_sum = lax.reduce_window(
                x, 0., lax.add, (1, self.pool_window, self.pool_window, 1),
                (1, self.strides, self.strides, 1), self.padding
            )
            if self.count_include_pad:
                div_term = np.square(self.pool_window)
            else:
                div_term = lax.reduce_window(
                    jnp.ones((1, h, w, 1)), 0., lax.add, (1, self.pool_window, self.pool_window, 1),
                    (1, self.strides, self.strides, 1), self.padding
                )
            x_pooled = x_sum / div_term
            return x_pooled

**Comment by cgarciae** [comment]
Hey @dslisleedh, additive pooling over 1s with 0-padding does seem like an easy way to create the counts, while there is probably a more efficient way to generate this "counts mask", we could probably use this implementation as a starting point. Curious what @marcvanzee and @jheek think about this?

**Comment by marcvanzee** [comment]
Thanks for proposing this @dslisleedh! I think this a good solution, it is exactly what I had in mind as well. Doing it more efficiently will most likely become quite complex so I recommend using this implementation and counting on XLA to do some optimizations for us if possible.

**Comment by marcvanzee** [comment]
@dslisleedh do you want to file a PR?

**Comment by dslisleedh** [comment]
Sure! Thank you for considering my suggestion and editing my comment to make it more readable. I'll make my request soon
